cmd/clusterctl/client/init.go

```diff
@@ -197,11 +204,16 @@ func (c *clusterctlClient) InitImages(options InitOptions) ([]string, error) {
 func (c *clusterctlClient) setupInstaller(cluster cluster.Client, options InitOptions) (cluster.ProviderInstaller, error) {
 	installer := cluster.ProviderInstaller()
 
+	var noKindMatchErr *meta.NoKindMatchError
 	providerList, err := cluster.ProviderInventory().List()
-	if err != nil {
+	if !errors.As(err, &noKindMatchErr) {
 		return nil, err
 	}
 
+	if providerList == nil {
+		providerList = &clusterctlv1.ProviderList{}
```

Comment:
How about doing `providerList = &clusterctlv1.ProviderList{}` before you do the `cluster.ProviderInventory().List()` call. The code would be cleaner that way.